src/scripts/thumbnail.py

```diff
@@ -0,0 +1,179 @@
+                        project=project,
+                        initial_view_state=user_project[0].initial_view_state,
+                        layers_project=layers_project,
+                        file_name=str(project.id)
+                        + project.updated_at.strftime("_%Y-%m-%d_%H-%M-%S-%f")
+                        + ".png",
+                    )
+
+                    # Update project with thumbnail url bypassing the model to avoid the table getting a new updated at
+                    await async_session.execute(
+                        text(
+                            """UPDATE customer.project
+                            SET thumbnail_url = :thumbnail_url WHERE id = :id""",
+                        ),
+                        {"thumbnail_url": thumbnail_url, "id": project.id},
+                    )
+                    await async_session.commit()
+
+                    # Delete old thumbnail from s3 if the thumbnail is not the default thumbnail
+                    if (old_thumbnail_url and settings.THUMBNAIL_DIR_PROJECT in old_thumbnail_url):
+                        settings.S3_CLIENT.delete_object(
+                            Bucket=settings.AWS_S3_ASSETS_BUCKET,
+                            Key=old_thumbnail_url.replace(
+                                settings.ASSETS_URL + "/", ""
+                            ),
+                        )
+                except Exception as e:
+                    print(f"Error updating project thumbnail: {e}")
+
+        # Fetch next page of projects to process
+        page += 1
+        projects = (await fetch_projects_to_update(async_session, last_run, page)).items
+
+
+async def process_layers(async_session: AsyncSession, last_run: datetime):
+    """Update thumbnails of layers."""
+
+    # Process projects page-by-page
+    page = 1
+    layers = (await fetch_layers_to_update(async_session, last_run, page)).items
+    while len(layers) > 0:
+        for layer in layers:
+            if (layer.type in (LayerType.feature, LayerType.table)):
+                try:
+                    old_thumbnail_url = layer.thumbnail_url
+
+                    # Create thumbnail
+                    print_map = PrintMap(async_session)
+                    thumbnail_url = await print_map.create_layer_thumbnail(
+                        layer=layer,
+                        file_name=str(layer.id) + "_" + str(uuid4()) + ".png",
+                    )
+
+                    # Update layer with thumbnail url bypassing the model to avoid the table getting a new updated at
+                    await async_session.execute(
+                        text(
+                            """UPDATE customer.layer
+                            SET thumbnail_url = :thumbnail_url WHERE id = :id""",
+                        ),
+                        {"thumbnail_url": thumbnail_url, "id": layer.id},
+                    )
+                    await async_session.commit()
```

Comment:
I think here we can avoid using raw sql. We have the thumbnail_url as property of layer model. We can update this property inside the loop and after the loop is finished we can say async_session.commit() which will push all "layer" object changes at once in the db.

Reply:
Hey Majk, thanks for the comments!

Reply:
I think the reason Elias did it this way is because updating the property in the model would cause the "updated_at" value of the project to also be set. Since we don't want to do this during a thumbnail update, a direct SQL approach was used.

But let me know if I've misunderstood something or you still feel we should update this via the model.